src/utils/actions.js

```diff
@@ -66,8 +66,83 @@ const baseIgActions = {
     // Through the source client?
     return context;
   },
-  'create-report': context => {
+  'create-report': async context => {
     // the output might define a profile
+    const action = context.action;
+
+    if (action.input) {
+      // input field defines which resource types to include
+
+      for (const input of action.input) {
+        const resourceType = input.type;
+        // ASSUMPTION: only types that are linked to a patient will be included here
+        // so for example Substance resources, which represent a Thing rather than an activity
+        // should not be here
+        const params = [];
+
+        // worth making this a switch instead of if?
+        if (resourceType === 'Patient') {
+          if (context.patient) {
+            // we already added the patient resource directly
+            // (see reporting_workflow:initializeContext)
+            continue;
+          } else {
+            // assume we have context.encounter
+            const patientId = context.encounter.subject.reference;
+            const patient = (await readFromEHR(patientId)).data;
+            context.records.push(patient);
+            // skip the rest of the logic below
+            continue;
+          }
+        } else if (resourceType === 'Encounter') {
+          if (context.encounter) {
+            // we already added the encounter resource directly
+            // (see reporting_workflow:initializeContext)
+            // ASSUMPTION: if given an encounter context,
+            //   we only care about that one, not all encounters
+            continue;
+          } else {
+            // assume we have context.patient
+            params.push(`subject=Patient/${context.patient.id}`);
+          }
+        } else {
+          // TODO: logic to check which resources actually have these fields?
+          // will be necessary if we get away from the simple US core types
```

Comment:
This is a good point. For the connectathon it looks like this is fine but definitely something we should look into (and possibly create a new task for the backlog if required)

Reply:
Created MEDMORPH-121 for this